staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -180,7 +180,6 @@ func NewCmdApply(baseName string, f cmdutil.Factory, ioStreams genericclioptions
 	}
 
 	// bind flag structs
-	o.DeleteFlags.AddFlags(cmd)
```

Comment:
The fix here is a bit more complicated than this because the `DeleteFlags` are where `-f` and `-k` come from.

https://github.com/kubernetes/kubernetes/blob/c8ebc8ab75a9c36453cf6fa30990fd0a277d856d/staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_flags.go#L35

https://github.com/kubernetes/kubernetes/blob/c8ebc8ab75a9c36453cf6fa30990fd0a277d856d/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/filename_flags.go#L34-L36

Reply:
I agree, it needs a refactor to separate the flags.

Reply:
Additionally, since we've added these flags here (even if by mistake) we need to ensure we don't break people suddenly.  So you'd need to copy these flags directly here, deprecate them, which will also hide them. This way ppl won't be broken when using these flags, especially in scripts, but will get clear information about its deprecation and that they don't do anything.